### ISSUE TITLE
Get any component from an entity

### ISSUE THREAD

**Issue by rodya-mirov** [slide-rs/specs#605]
## Description
This would allow grabbing component values from an entity, or list of entities. Ideally if we had an iterable of `Entity`, we could join it with components.

Note that this appears to be a duplicate of https://github.com/slide-rs/specs/issues/362 ; however the discussion on that issue concerned scripting, which is much more complicated, and eventually closed. This calls for implementing the actual text at the top of that issue, instead of the motivation for that issue (which is more complicated).

## Motivation
I am implementing a roguelike-style architecture, and I already know I want to print (e.g.) squares with position x in 10..50, y in 20..60. Iterating through all the squares and filtering is annoying and not performant. Similarly, checking if a certain square is occupied is annoying; I have the coordinates, but I can't look up that square in O(1), I have to loop through everything.

## Drawbacks
Given how obvious this feature seems to be, I have to assume it was intentionally not included, and perhaps it is counter to the design of ECS systems in general. I'm not sure what the intended flow is, for large numbers of entities, to efficiently query within them. Several notes in the documentation talk about "keeping a cache of entities" or etc., but it seems like you can't really do anything with an entity from within a system.

I do not see this as a breaking change.

## Unresolved questions

Basically, see drawbacks. It's possible there's a more "idiomatic" way to achieve these things, but it's not clear to me what it is. So maybe just more documentation (in the book) about the "right way" to use these entity caches, etc. would be more appropriate.

---

I can try to work on this but I'm not sure I understand the underlying design enough to do what's needful. Happy to try.

**Comment by rodya-mirov** [comment]
Sorry, I see now that I am not a good reader. The book does not mention it, but the `Storage` types (e.g. ReadStorage, WriteStorage) have a `.get(&self, entity:Entity) -> Option<&T>` method that does exactly what I was asking for.
